### PR TITLE
Implementing initial datastructures

### PR DESCRIPTION
**Pull request by JohnHynes**
### **Summary**

     - Classes.java
        - Manages and interfaces with Entities and Relationships

     - Relationship.java
        - Contains data relating to relationships and their expected usage

     - Entity.java
        - Contains data and methods relating to entities and their expected usage

### CODE REVIEW

File: src/Classes.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			entities.get(index).name = newname;
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	public boolean deleteClass(String target) {
+		return entities.remove(new Entity(target));
+	}
```
Comment (phila413) [inline src/Classes.java:39]: Shouldn't this also delete any relationships tied to the class as well?

Comment (JohnHynes) [reply src/Classes.java:39]: That's a great find, Ian and I didn't implement the relationship functions yet so whenever we add those in we can definitely add that in as well.

File: src/Entity.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	//********************************************************//
+	// Attribute Functions                                    //
+	//********************************************************//
+	
+	
+	public boolean createAttribute(String attribute) {
+		return attributes.add(attribute);
+	}
```
Comment (HomelessSkittle) [inline src/Entity.java:42]: Are we allowing multiple attributes of the same name?

Comment (JohnHynes) [reply src/Entity.java:42]: We can add in a check for duplicates

**Review by SerbianSensation — APPROVED** [review]

**Pull request merged**